velox/functions/lib/string/StringImpl.h

```diff
@@ -288,6 +288,18 @@ FOLLY_ALWAYS_INLINE void sha512(TOutString& output, const TInString& input) {
       folly::ByteRange((const uint8_t*)input.data(), input.size()));
 }
 
+// Compute the HMAC-SHA256 Hash.
+template <typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE bool
+HmacSha256(TOutString& output, const TInString& key, const TInString& data) {
+  output.resize(32);
+  folly::ssl::OpenSSLHash::hmac_sha256(
```

Comment:
Is this calculation inline with presto java side implementation: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/HmacFunctions.java#L50? Seems to be different from code wise. Could you also confirm by running presto java as you did in test with python lib to verify? Thanks!

Reply:
I tried one query and it seems matching:

```
presto:de> select to_hex(hmac_sha256(to_utf8('hashme'), to_utf8('velox')));
                              _col0
------------------------------------------------------------------
 24BB7FA25FD592EF6A4C939D4FB91B7F7F04F8813260961101117EC30F865794
```

Reply:
@Yuhta Thanks for you help.

Reply:
> Is this calculation inline with presto java side implementation: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/scalar/HmacFunctions.java#L50? Seems to be different from code wise. Could you also confirm by running presto java as you did in test with python lib to verify? Thanks!

@xiaoxmeng I use the expected values from the presto's [hmac sha256 test case](https://github.com/prestodb/presto/blob/master/presto-main/src/test/java/com/facebook/presto/operator/scalar/TestVarbinaryFunctions.java#L362:~:text=assertFunction(%22-,sha256,-(CAST(%27%27%20AS%20VARBINARY), which use the `"key"` as the *key*, and the result matches. I also use add the expected values in the presto's test case, and results match too. IMHO they should follow the same standard.
```Java
@Test
    public void testHmacSHA256()
    {
        assertFunction("hmac_sha256(CAST('' AS VARBINARY), CAST('key' AS VARBINARY))",
                VARBINARY, sqlVarbinaryHex("5D5D139563C95B5967B9BD9A8C9B233A9DEDB45072794CD232DC1B74832607D0"));
        assertFunction("hmac_sha256(CAST('hashme' AS VARBINARY), CAST('key' AS VARBINARY))",
                VARBINARY, sqlVarbinaryHex("D3D72F9FACDE059DA3A4EB43A9ABDD4B35118E0FEF00E6D16FB04BB332AF0484"));
        assertFunction("hmac_sha256(CAST('' AS VARBINARY), CAST('velox' AS VARBINARY))",
                VARBINARY, sqlVarbinaryHex("fd8658b6a6b6601155fecf9a39b6f95cf030863e550073423a8e250a35c6f5a4"));
        assertFunction("hmac_sha256(CAST('hashme' AS VARBINARY), CAST('velox' AS VARBINARY))",
                VARBINARY, sqlVarbinaryHex("24bb7fa25fd592ef6a4c939d4fb91b7f7f04f8813260961101117ec30f865794"));
    }
```
```C++
EXPECT_EQ(
      hexToDec(
          "D3D72F9FACDE059DA3A4EB43A9ABDD4B35118E0FEF00E6D16FB04BB332AF0484"),
      hmacSha256("hashme", "key"));
  
  EXPECT_EQ(
      hexToDec(
          "5D5D139563C95B5967B9BD9A8C9B233A9DEDB45072794CD232DC1B74832607D0"),
      hmacSha256("", "key"));
```

Reply:
@duanmeng @Yuhta thanks for the verification!